semantic/monotype.go

```diff
@@ -442,20 +425,226 @@ func (l MonoType) Equal(r MonoType) bool {
 	return false
 }
 
+func newBasicType(t fbsemantic.Type) MonoType {
+	builder := flatbuffers.NewBuilder(16)
+	offset := buildBasicType(builder, t)
+	builder.Finish(offset)
+
+	buf := builder.FinishedBytes()
+	basic := fbsemantic.GetRootAsBasic(buf, 0)
+	mt, err := NewMonoType(basic.Table(), fbsemantic.MonoTypeBasic)
+	if err != nil {
+		panic(err)
+	}
+	return mt
+}
+
 func NewArrayType(elemType MonoType) MonoType {
```

Comment:
Can you comment this method, and also indicate that it will panic and under what conditions?

Reply:
There's a panic in there, but it's from `NewMonoType` which is getting invoked in a way where it can never panic. So the panic is more to catch that something is really wrong. I'll add a godoc though.